### ISSUE TITLE
Show dates in the campaign selection screen?

### ISSUE THREAD

**Issue by jostephd** [wesnoth/wesnoth#4212]
Anybody interested in adding the `start_year` of a campaign to the campaign listing, like this?

![2019-08-11-001012_1010x659_scrot](https://user-images.githubusercontent.com/24784687/62828055-a061d780-bbcc-11e9-8c9c-5b700b3d977e.png)

**Comment by Vultraz** [comment]
I don't think that's a great idea. Without a frame of reference, the Average player would have no idea what those numbers mean (or even what YW means).

**Comment by jostephd** [comment]
Good point, I agree: we don't want to present unnecessary information to casual players. And it's not just YW: TRoW and UtBS's start dates are "2 BW" and "300 AF" respectively.

For context, I think the reason I bothered to implement that was because I figured the start_date was lore and it wasn't presented anywhere else in the UI.

**Comment by Pentarctagon** [comment]
Would it be possible to have the year value be right justified, so it shows up under the Dates column?  Currently it allows sorting chronologically, but doesn't show the actual dates it's sorting by.

**Comment by jostephd** [comment]
> Would it be possible to have the year value be right justified, so it shows up under the Dates column?

Not easily. It's not a table, the "Dates" and the list of campaigns are two separate widgets.

> doesn't show the actual dates it's sorting by.

Maybe the dates could be shown in the large pane on the right. Or maybe we could add a "More information" button that pops up another dialog, leaving the main campaign selection dialog streamlined. If we had such a place to dump extra information, we could use it to show the forum thread link, campaign version (if UMC), and so on, so this information wouldn't clutter the campaign selection screen

**Comment by Vultraz** [comment]
It could be done, same as any other dialog that has column-based sorting, but I think it would just clutter the UI more.

**Comment by CelticMinstrel** [comment]
Another possibility would be to show it only when sorting by date...

**Comment by jostephd** [comment]
Yet another possibility: add the campaign name as a header and below it in small print the start year, like this:

![2019-08-26-170022_964x348_scrot](https://user-images.githubusercontent.com/24784687/63708097-01bbb480-c823-11e9-91fb-92b6056acc50.png)

**Comment by jostephd** [comment]
> Yeah, that could work too. Note that AToTB also ends in the same year it begins, so I wonder if we'd want that to be clear from the text?

I think showing the end year could be a spoiler, or possibly wrong (if there are forks in the campaign, as in TSG).

Also, not to nitpick but to clarify, in AToTB the combat scenarios all happen in 363YW, like `_main.cfg` says, but the epilogue scenario happens in 365YW.

**Comment by Wedge009** [comment]
I would be in favour of seeing the year, despite the potential for confusion. Personally I think it's easy to guess YW is Year of Wesnoth. In contrast, _Final Fantasy Type-0_ has a calendar system designating years as RG. To this day I still have no clue what RG is supposed to mean (though I appreciate the lore).

A potential issue I see is maintenance - lore discussion on the forums sometimes results in slight adjustments to the time-line (seen this in PRs from a few months back). Maybe I missed it in the discussion here, but is the display simply exposing what is contained internally with respect to the sort-by-date option? Or is this something we have to maintain separately for the individual campaigns?

**Comment by soliton-** [comment]
The dates are in the campaign WML just not shown anywhere so far.

**Comment by cooljeanius** [comment]
> The dates are in the campaign WML just not shown anywhere so far.

Yeah I think exposing them to players in the UI could be helpful to prompt add-on authors to be sure to set the dates in their own campaigns, too.

**Comment by Gothyoba** [comment]
> I don't think that's a great idea. Without a frame of reference, the average player would have no idea what those numbers mean (or even what YW means).

YW, BW and AF are now explained in game in the calendar section of the encyclopedia.

**Comment by cooljeanius** [comment]
Thread on the forums wishing for this: https://forums.wesnoth.org/viewtopic.php?t=59064
